tests/unit/test_backends.py

```diff
@@ -12,179 +12,142 @@
-        responseStr = self._backend.runListReferenceBases(id_, requestArgs)
-        response = protocol.ListReferenceBasesResponse.fromJsonString(
-            responseStr)
-        self.assertTrue(
-            isinstance(response, protocol.ListReferenceBasesResponse))
-
-    def testRunListReferenceBases(self):
-        referenceSet = self._backend.getReferenceSets()[0]
-        reference = referenceSet.getReferences()[0]
-        self.runListReferenceBases(reference.getId())
-
-    def testDatasetNotFound(self):
-        request = protocol.SearchVariantSetsRequest()
-        datasetId = 'doesNotExist'
-        request.datasetId = datasetId
-        with self.assertRaises(exceptions.DatasetNotFoundException):
-            self._backend.getDataset(request.datasetId)
-
-
-class TestFileSystemBackend(TestAbstractBackend):
+            self._backend.getReferenceSetByName(referenceSet.getLocalId()),
+            referenceSet
+        )
+        self.assertEqual(self._backend.getReferenceSet(referenceSet.getId()),
+                         referenceSet)
+
+    def testAddMultipleReferenceSet(self):
+        firstRSLocalId = "id1"
+        firstRS = references.AbstractReferenceSet(firstRSLocalId)
+        secondRSLocalId = "id2"
+        secondRS = references.AbstractReferenceSet(secondRSLocalId)
+        self.assertEqual(self._backend.getNumReferenceSets(), 0)
+        self.assertEqual(self._backend.getReferenceSets(), [])
+        self._backend.addReferenceSet(firstRS)
+        self._backend.addReferenceSet(secondRS)
+        self.assertEqual(self._backend.getNumReferenceSets(), 2)
+        self.assertEqual(self._backend.getReferenceSets(),
+                         [firstRS, secondRS])
+        self.assertEqual(self._backend.getReferenceSetByIndex(0),
+                         firstRS)
+        self.assertEqual(self._backend.getReferenceSetByIndex(1),
+                         secondRS)
+        self.assertEqual(
+            self._backend.getReferenceSetByName(firstRS.getLocalId()),
+            firstRS
+        )
+        self.assertEqual(
+            self._backend.getReferenceSetByName(secondRS.getLocalId()),
+            secondRS
+        )
+        self.assertEqual(self._backend.getReferenceSet(firstRS.getId()),
+                         firstRS)
+        self.assertEqual(self._backend.getReferenceSet(secondRS.getId()),
+                         secondRS)
+
+    def testGetDatasetBadId(self):
+        for badId in ["", None, "NO SUCH ID"]:
+            self.assertRaises(
+                exceptions.DatasetNotFoundException,
+                self._backend.getDataset, badId)
+
+    def testGetReferenceSetBadId(self):
+        for badId in ["", None, "NO SUCH ID"]:
+            self.assertRaises(
+                exceptions.ReferenceSetNotFoundException,
+                self._backend.getReferenceSet, badId)
+
+    def testGetDatasetBadName(self):
+        for badName in ["", None, "NO SUCH ID"]:
```

Comment:
The pedant in me wants to change this to "NO SUCH NAME", but obviously it makes no difference.